importer/tasks.py

```diff
@@ -3,11 +3,55 @@
 """
 from __future__ import unicode_literals
 
+import logging
+
+from django.conf import settings
+
 from lore.celery import async
-from importer.api import import_course_from_file
+from learningresources.api import update_xanalytics
+from xanalytics import send_request, get_result, get_xanalytics_file
+
+log = logging.getLogger(__name__)
 
 
 @async.task
 def import_file(path, repo_id, user_id):
     """Asynchronously import a course."""
+    from importer.api import import_course_from_file
     import_course_from_file(path, repo_id, user_id)
+
+
+@async.task
+def populate_xanalytics_fields(course_id):
+    """
+    Initiate request to xanalytics API to get stats for a course,
+    then trigger async job to retrieve results when they become available.
+    Args:
+        course_id (int): primary key of a Course
+    """
+    if settings.XANALYTICS_URL != "":
+        token = send_request(settings.XANALYTICS_URL + "/create", course_id)
+        check_for_results.apply_async(
+            kwargs={"token": token, "wait": 5, "attempt": 1}, countdown=5)
+
+
+@async.task
+def check_for_results(token, wait, attempt):
+    """
+    Check for xanalytics results for a course.
+    Args:
+        token (string): Token received from xanalytics server.
+        wait (int): Seconds to wait before repository.
+        attempt (int): Attempt number, so we don't try forever.
+    """
+    resp = get_result(settings.XANALYTICS_URL + "/status", token)
+    if resp["status"] == "still busy" and attempt < 10:
```

Comment:
I think the "potentially forever" thing would be a problem; we would never want that to happen. It _should_ log if it gives up, though. Preferably sending an e-mail alert. If it went on forever it would increasingly tax the remote API, and also just consume resources on our servers. Also, this is something we could easily trigger from outside the import task, so it would be easy to "backfill" these once deployed, and also easy to make another attempt if we were alerted it had failed.

The retry count being a "constant" is definitely the way to go here.